src/modules/common/containers/LanguageSelector.js

```diff
@@ -27,33 +36,71 @@ export class LanguageSelector extends React.Component {
    * @param languageCode The languageCode
    * @return {string} The path
    */
-  getPathForLanguage (languageCode) {
-    return this.props.languageChangeUrls[languageCode] || `/${this.props.location}/${languageCode}`
+  getLanguageChangeAction (languageCode) {
+    const {location, categories, events} = this.props
+    const {city, eventId, extraAlias} = location.payload
+    const routeType = location.type
+
+    switch (routeType) {
+      case CATEGORIES_ROUTE:
+        if (categories) {
+          try {
+            const category = categories.getCategoryByUrl(location.pathname)
+            if (category && category.id !== 0) {
+              return goToCategoriesRedirect(city, languageCode, `${category.availableLanguages[languageCode]}`)
+            }
+          } catch (e) {
```

Comment:
What exception does this catch? Can flow add types here?

Reply:
Apparently not :(, https://github.com/facebook/flow/issues/2470

Reply:
Where is this exception thrown? Exceptions should be used for the handling of unexpected errors. This more sounds like another return value.

Reply:
in getCategoriesByUrl. We can remove the exception there, but getAncestors will be ugly than because of flow